src/routes/search/posts/+page.svelte

```diff
@@ -1,64 +1,90 @@
 <script lang="ts">
-	import { Toast, getToastStore } from '@skeletonlabs/skeleton';
+	import { getToastStore } from '@skeletonlabs/skeleton';
 	import Icon from '@iconify/svelte';
 	import { onMount } from 'svelte';
 	import { get } from 'svelte/store';
-	import { token, serverURL } from '$lib/Store';
+	import { token } from '$lib/Store';
 	import { goto } from '$app/navigation';
 	import { t } from '../../../i18n';
+	import type { Feed as FeedStructure } from '$lib/types/Feed';
 	import Feed from '../../../components/Feed.svelte';
-	import { resetPostsAndFeedData, searchHashtagPosts } from '$lib/FeedFunctions';
-	import { posts, feedData, pageType } from '$lib/FeedDataStore';
-	import { onDestroy } from 'svelte';
-
-	onDestroy(() => {
-		posts.set([]);
-		feedData.set({
-			records: [],
-			pagination: {
-				lastPostId: '',
-				limit: 10,
-				records: 0
-			}
-		});
-	});
-	let prevInput = '';
-	let characterRemoved = false;
+	import { searchPostsByHashtag } from './searchPosts';
 
+	let hasMorePosts = true;
+	let maxPostCounter = 0;
+	let lastInput = '';
+	let slotLimit = 2;
+	let feedData: FeedStructure = {
+		records: [],
+		pagination: {
+			lastPostId: '',
+			limit: slotLimit,
+			records: 0
+		}
+	};
 	const toastStore = getToastStore();
-
-	const serverUrl = get(serverURL);
+	const loginToken = get(token);
 
 	async function handleHashtagInput(event: Event) {
 		let hashtagInput = (event.target as HTMLInputElement).value;
-
-		if (hashtagInput.length > prevInput.length) {
-			characterRemoved = false;
-		} else if (hashtagInput.length < prevInput.length) {
-			characterRemoved = true;
+		lastInput = hashtagInput;
+		// resetting the feed data before each new search
+		hasMorePosts = true;
+		maxPostCounter = 0;
+		if (hashtagInput.length > 0) {
```

Comment:
brauchst du nicht wenn das ding leer ist kommt automatisch ein leeres Objekt zurück

Reply:
Doch braucht man weil bei jedem neuen Input ein komplett neuer Request gesendet wird.
Beispiel:
Suche zuerst nach p
Ergebnis 1: Post
Ergebnis 2: Past
Gib pa ein:
jetzt wird in den Ergebnissen immer noch Post dargestellt.

Deswegen muss vor jedem neuen Request alles geresettet werden.